pulsar/backoff/backoff.go

```diff
@@ -15,7 +15,7 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package internal
+package backoff
```

Comment:
Could you move this pack to the `pulsar` directory and keep `BackoffPolicy`?

Reply:
This will lead to a circular dependency.

https://github.com/apache/pulsar-client-go/blob/master/pulsar/internal/http_client.go#L151